esmvaltool/diag_scripts/autoassess/autoassess_stratosphere.py

```diff
@@ -0,0 +1,159 @@
+"""autoassess stratosphere diagnostic."""
+import os
+import logging
+import inspect
+import sys
+import subprocess
+import numpy as np
+
+import iris
+import yaml
+
+logger = logging.getLogger(__name__)
+
+# Diagnostic that takes two models (control_model and exp_model
+# and observational data (ERA-Interim and MERRA)
+
+
+def _check_p30(cubelist):
+    """Check for plev = 30hPa"""
+    p30 = iris.Constraint(air_pressure=3000.)
+    for cube in cubelist:
+        qbo30 = cube.extract(p30)
+        # extract the masked array and check for missing data
+        qbo30.data = np.ma.array(qbo30.data)
+        if np.all(qbo30.data.mask):
+            logger.info('Cube metadata:', qbo30.metadata)
+            logger.error('All data is masked at 30hPa! Exiting.')
+            sys.exit(1)
+
+
+def get_cfg():
+    """Read diagnostic script configuration from settings.yml."""
+    settings_file = sys.argv[1]
+    with open(settings_file) as file:
+        cfg = yaml.safe_load(file)
+    return cfg
+
+
+def get_input_files(cfg):
+    """Get a dictionary list from metadata.yml files."""
+    metadata_files = cfg['input_files']
+    metadatas = []
+    for metadata_file in metadata_files:
+        with open(metadata_file) as file:
+            metadata = yaml.safe_load(file)
+            metadatas.append(metadata)
+    return metadatas
+
+
+def main():
+    """Execute the stratosphere area"""
+    cfg = get_cfg()
```

Comment:
I would recommend getting the configuration and list of input files as is done in the [example diagnostic](https://github.com/ESMValGroup/ESMValTool/blob/REFACTORING_backend/esmvaltool/diag_scripts/examples/diagnostic.py).